dex/elrond_dex_farm/src/lib.rs

```diff
@@ -447,6 +462,32 @@ pub trait Farm:
         })
     }
 
+    fn create_farm_tokens_by_merging(
+        &self,
+        amount: &mut Self::BigUint,
+        token_id: &TokenIdentifier,
+        attributes: &mut FarmTokenAttributes<Self::BigUint>,
+        caller: &Address,
+    ) -> SCResult<Nonce> {
+        let current_position_replic = FarmToken {
+            token_amount: GenericEsdtAmountPair {
+                token_id: token_id.clone(),
+                token_nonce: 0,
+                amount: amount.clone(),
+            },
+            attributes: attributes.clone(),
+        };
+
+        let merged_attributes =
```

Comment:
we might think about sending directly the earned rewards till that moment. it would be easier to code as well. Thus on enter farm, when we merge, we could actually do exitFarm with the old position, enter farm with the OLD + NEW position and SEND rewards to the user. Or directly compound (although an option) if we are in the FARM MEX to MEX

Reply:
i don't think it would be easier to code. why is this way not good?

Reply:
it is ok. I was thinking only about the alternatives and about what the users want. We would save one transaction when the user wants to addToFarm and harvest as well. It can stay like this, however we might want further improvements which makes it easier for users to interact with the contracts, and makes them click less buttons, less transactions.

Reply:
i understand but i'd say we also have to limit somehow the actions we do in one transaction, gas issues are hard to deal with and we have also contracts interacting with each other. one more endpoint in the farm means one more endpoint in the proxy